### PR TITLE
Remove Default Request Hardcoded Aliases

### PR DESCRIPTION
**Pull request by SyntaxNode**
As we move to the next major Prebid Server release, it's time to remove legacy behavior where hardcoded aliases were "promoted" from the default request to the server level. 

Support for hardcoded request aliases has been fully implemented in Prebid Server for a while. The legacy approach was a quick way to provide a limited ability for hardcoded aliases. Now that we have the full implementation, it's time for this to go.

**If you are host currently using this feature** you will need to add a yaml file defining the alias instead for the alias to appear in the server bidder list. Aliases defined in the default request will continue to work as normal request level aliases without change.

### CODE REVIEW

File: endpoints/info/bidders_test.go [hunk 1]
```diff
@@ -19,88 +19,60 @@ func TestPrepareBiddersResponseAll(t *testing.T) {
...
+			expected:     `["a","z"]`,
 		},
 		{
-			name:                "mixed",
-			givenBidders:        config.BidderInfos{"a": disabledCore, "b": disabledAlias, "c": enabledCore, "d": enabledAlias},
-			givenRequestAliases: map[string]string{"z": "a"},
-			expected:            `["a","b","c","d","z"]`,
+			name:         "mixed",
+			givenBidders: config.BidderInfos{"a": disabledCore, "b": disabledAlias, "c": enabledCore, "d": enabledAlias},
+			expected:     `["a","b","c","d"]`,
```
Comment (bsardo) [inline endpoints/info/bidders_test.go:69]: I think you can delete this test case. It is the same as the `alias-many-mixed` case on lines 57-59.

File: endpoints/info/bidders_test.go [hunk 2]
```diff
@@ -155,6 +127,11 @@ func TestPrepareBiddersResponseAllBaseOnly(t *testing.T) {
 			givenBidders: config.BidderInfos{"a": disabledCore, "b": disabledAlias, "c": enabledCore, "d": enabledAlias},
 			expected:     `["a","c"]`,
 		},
+		{
+			name:         "mixed",
+			givenBidders: config.BidderInfos{"a": disabledCore, "b": disabledAlias, "c": enabledCore, "d": enabledAlias},
+			expected:     `["a","c"]`,
+		},
```
Comment (bsardo) [inline endpoints/info/bidders_test.go:134]: I think you can delete this test case. It is the same as the `alias-mixed` case on lines 126-128.

File: endpoints/info/bidders_test.go [hunk 3]
```diff
@@ -175,88 +152,60 @@ func TestPrepareBiddersResponseEnabledOnly(t *testing.T) {
...
+			expected:     `["a","z"]`,
 		},
 		{
-			name:                "mixed",
-			givenBidders:        config.BidderInfos{"a": disabledCore, "b": disabledAlias, "c": enabledCore, "d": enabledAlias},
-			givenRequestAliases: map[string]string{"z": "a"},
-			expected:            `["c","d"]`,
+			name:         "mixed",
+			givenBidders: config.BidderInfos{"a": disabledCore, "b": disabledAlias, "c": enabledCore, "d": enabledAlias},
+			expected:     `["c","d"]`,
```
Comment (bsardo) [inline endpoints/info/bidders_test.go:202]: I think you can delete this test case. It is the same as the `alias-mixed` case on lines 190-192.

**Review by bsardo — COMMENTED** [review]
LGTM other than a couple of quick nitpicks. Local testing looks good as well.

File: router/router.go
[the remark this answers is not in the document]
Comment (SyntaxNode) [reply router/router.go:1]: Yes. That's merging in the default stored request, not specific to aliases. Not sure why the comments are indicating as such. Changing the code to not be alias specific.

@hhhjort you originally wrote this, any idea?

File: endpoints/info/bidders_test.go (hunk 1 above)
Comment (SyntaxNode) [reply endpoints/info/bidders_test.go:69]: Differentiated.

File: endpoints/info/bidders_test.go (hunk 2 above)
Comment (SyntaxNode) [reply endpoints/info/bidders_test.go]: Differentiated.

File: endpoints/info/bidders_test.go (hunk 3 above)
Comment (SyntaxNode) [reply endpoints/info/bidders_test.go:217]: Differentiated.

**Review by bsardo — APPROVED** [review]

**Pull request merged**